lib/block-supports/layout.php

```diff
@@ -173,6 +194,11 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 	$block_classname    = wp_get_block_default_classname( $block['blockName'] );
 	$container_class    = wp_unique_id( 'wp-container-' );
 	$layout_classname   = '';
+	$use_global_padding = isset( $used_layout['inherit'] ) && $used_layout['inherit'] || isset( $used_layout['contentSize'] ) && $used_layout['contentSize'] || isset( $used_layout['useGlobalPadding'] ) && $used_layout['useGlobalPadding'];
```

Comment:
Instead of the `useGlobalPadding` boolean, given that the setting is now deterministic, would it be possible to do a check for the theme.json global flag instead, to opt-in to this behaviour?

Reply:
It should work if we just check for `$used_layout['inherit']` or `isset( $used_layout['contentSize']` because those are the only circumstances in which we auto-apply the padding.